velox/expression/CastHooks.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/expression/FunctionCallToSpecialForm.h"
+#include "velox/expression/SpecialForm.h"
+#include "velox/type/Conversions.h"
+
+namespace facebook::velox::exec {
+class CastExpr;
+
+class CastHooks {
+ public:
+  virtual ~CastHooks() {}
+
+  static const std::shared_ptr<CastHooks>& get();
```

Comment:
> Two missing points are 1) support more Spark-allowed patterns for cast from string to timestamp (better to go into a separate PR). 2) handle white-spaces differently for cast from string to decimal (should be handled in #5307).

Would you extend the Hooks to add API that are needed to support the above? I'd like to make sure these APIs "fit in" nicely.

Reply:
For 1), no extra hook is needed. By using `removeWhiteSpaces` at https://github.com/facebookincubator/velox/pull/5307/files#diff-872249bbe84c54f0119d4568444009b9b850ec34b802ebbc0b641bb012fa02baR559, this issue should be fixed.
For 2), `castStringToTimestamp` is added. In the future implementation, an option could be added to `fromTimestampString`, and the spark hook could call it with customized option.